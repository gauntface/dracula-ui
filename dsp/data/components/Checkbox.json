{
  "dsp_spec_version": "0.0.1",
  "last_updated_by": "System",
  "last_updated": "2020-10-04T23:51:23.416Z",
  "entities": [
    {
      "class": "component",
      "type": "page",
      "id": "Checkbox",
      "name": "Checkbox",
      "last_updated": "2020-10-04T23:51:23.416Z",
      "last_updated_by": "System",
      "description": "Checkbox is a styled HTML Input of type checkbox.\n\nThere are no behavior changes applied to the native HTML tag other\nthan light styling done via CSS in order to keep check boxes accessible.",
      "related_entity_ids": [],
      "tags": ["component"],
      "snippets": {
        "trigger": "Checkbox",
        "languages": {
          "javascript": "<div style={{...}}>\n  <div style={{...}}>\n    <Checkbox id=\"one\" defaultChecked={true} color=\"purple\" name=\"demo\" />\n    <label htmlFor=\"one\" className=\"drac-text drac-text-white\">\n      Option 1\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"two\" color=\"purple\" name=\"demo\" />\n    <label htmlFor=\"two\" className=\"drac-text drac-text-white\">\n      Option 2\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"three\" color=\"purple\" name=\"demo\" />\n    <label htmlFor=\"three\" className=\"drac-text drac-text-white\">\n      Option 3\n    </label>\n  </div>\n</div>",
          "html": "<div style=\"display: flex; flex-direction: column; padding: 3rem\">\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"one\"\n      color=\"purple\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-purple\"\n      checked=\"\"\n    /><label for=\"one\" class=\"drac-text drac-text-white\">Option 1</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"two\"\n      color=\"purple\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-purple\"\n    /><label for=\"two\" class=\"drac-text drac-text-white\">Option 2</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"three\"\n      color=\"purple\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-purple\"\n    /><label for=\"three\" class=\"drac-text drac-text-white\">Option 3</label>\n  </div>\n</div>\n"
        }
      },
      "ext_com_draculaui_variations": [
        {
          "title": "Usage",
          "html": "<div style=\"display: flex; flex-direction: column; padding: 3rem\">\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"one\"\n      color=\"purple\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-purple\"\n      checked=\"\"\n    /><label for=\"one\" class=\"drac-text drac-text-white\">Option 1</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"two\"\n      color=\"purple\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-purple\"\n    /><label for=\"two\" class=\"drac-text drac-text-white\">Option 2</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"three\"\n      color=\"purple\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-purple\"\n    /><label for=\"three\" class=\"drac-text drac-text-white\">Option 3</label>\n  </div>\n</div>\n",
          "react": "<div style={{...}}>\n  <div style={{...}}>\n    <Checkbox id=\"one\" defaultChecked={true} color=\"purple\" name=\"demo\" />\n    <label htmlFor=\"one\" className=\"drac-text drac-text-white\">\n      Option 1\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"two\" color=\"purple\" name=\"demo\" />\n    <label htmlFor=\"two\" className=\"drac-text drac-text-white\">\n      Option 2\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"three\" color=\"purple\" name=\"demo\" />\n    <label htmlFor=\"three\" className=\"drac-text drac-text-white\">\n      Option 3\n    </label>\n  </div>\n</div>",
          "docs": ""
        },
        {
          "title": "Disabled",
          "html": "<div style=\"display: flex; flex-direction: column; padding: 3rem\">\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"one\"\n      disabled=\"\"\n      color=\"white\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-white\"\n      checked=\"\"\n    /><label for=\"one\" class=\"drac-text drac-text-white\">Checked</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"two\"\n      disabled=\"\"\n      color=\"white\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-white\"\n    /><label for=\"two\" class=\"drac-text drac-text-white\">Unchecked</label>\n  </div>\n</div>\n",
          "react": "<div style={{...}}>\n  <div style={{...}}>\n    <Checkbox id=\"one\" disabled={true} defaultChecked={true} color=\"white\" name=\"demo\" />\n    <label htmlFor=\"one\" className=\"drac-text drac-text-white\">\n      Checked\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"two\" disabled={true} color=\"white\" name=\"demo\" />\n    <label htmlFor=\"two\" className=\"drac-text drac-text-white\">\n      Unchecked\n    </label>\n  </div>\n</div>",
          "docs": ""
        },
        {
          "title": "Colors",
          "html": "<div style=\"display: flex; flex-direction: column; padding: 3rem\">\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"black\"\n      color=\"black\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-black\"\n      checked=\"\"\n    /><label for=\"black\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"white\"\n      color=\"white\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-white\"\n      checked=\"\"\n    /><label for=\"white\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"cyan\"\n      color=\"cyan\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-cyan\"\n      checked=\"\"\n    /><label for=\"cyan\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"green\"\n      color=\"green\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-green\"\n      checked=\"\"\n    /><label for=\"green\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"orange\"\n      color=\"orange\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-orange\"\n      checked=\"\"\n    /><label for=\"orange\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"pink\"\n      color=\"pink\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-pink\"\n      checked=\"\"\n    /><label for=\"pink\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"purple\"\n      color=\"purple\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-purple\"\n      checked=\"\"\n    /><label for=\"purple\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"red\"\n      color=\"red\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-red\"\n      checked=\"\"\n    /><label for=\"red\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"yellow\"\n      color=\"yellow\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-yellow\"\n      checked=\"\"\n    /><label for=\"yellow\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"animated\"\n      color=\"animated\"\n      name=\"demo\"\n      class=\"drac-checkbox\"\n      checked=\"\"\n    /><label for=\"animated\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"cyanGreen\"\n      color=\"cyanGreen\"\n      name=\"demo\"\n      class=\"drac-checkbox\"\n      checked=\"\"\n    /><label for=\"cyanGreen\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"pinkPurple\"\n      color=\"pinkPurple\"\n      name=\"demo\"\n      class=\"drac-checkbox\"\n      checked=\"\"\n    /><label for=\"pinkPurple\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"purpleCyan\"\n      color=\"purpleCyan\"\n      name=\"demo\"\n      class=\"drac-checkbox\"\n      checked=\"\"\n    /><label for=\"purpleCyan\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"yellowPink\"\n      color=\"yellowPink\"\n      name=\"demo\"\n      class=\"drac-checkbox\"\n      checked=\"\"\n    /><label for=\"yellowPink\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n</div>\n",
          "react": "<div style={{...}}>\n  <div style={{...}}>\n    <Checkbox id=\"black\" defaultChecked={true} color=\"black\" name=\"demo\" />\n    <label htmlFor=\"black\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"white\" defaultChecked={true} color=\"white\" name=\"demo\" />\n    <label htmlFor=\"white\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"cyan\" defaultChecked={true} color=\"cyan\" name=\"demo\" />\n    <label htmlFor=\"cyan\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"green\" defaultChecked={true} color=\"green\" name=\"demo\" />\n    <label htmlFor=\"green\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"orange\" defaultChecked={true} color=\"orange\" name=\"demo\" />\n    <label htmlFor=\"orange\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"pink\" defaultChecked={true} color=\"pink\" name=\"demo\" />\n    <label htmlFor=\"pink\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"purple\" defaultChecked={true} color=\"purple\" name=\"demo\" />\n    <label htmlFor=\"purple\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"red\" defaultChecked={true} color=\"red\" name=\"demo\" />\n    <label htmlFor=\"red\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"yellow\" defaultChecked={true} color=\"yellow\" name=\"demo\" />\n    <label htmlFor=\"yellow\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"animated\" defaultChecked={true} color=\"animated\" name=\"demo\" />\n    <label htmlFor=\"animated\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"cyanGreen\" defaultChecked={true} color=\"cyanGreen\" name=\"demo\" />\n    <label htmlFor=\"cyanGreen\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"pinkPurple\" defaultChecked={true} color=\"pinkPurple\" name=\"demo\" />\n    <label htmlFor=\"pinkPurple\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"purpleCyan\" defaultChecked={true} color=\"purpleCyan\" name=\"demo\" />\n    <label htmlFor=\"purpleCyan\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{...}}>\n    <Checkbox id=\"yellowPink\" defaultChecked={true} color=\"yellowPink\" name=\"demo\" />\n    <label htmlFor=\"yellowPink\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n</div>",
          "docs": ""
        }
      ],
      "ext_com_draculaui_props": {
        "color": {
          "defaultValue": null,
          "description": "The Dracula UI theme color for the checkbox.",
          "name": "color",
          "parent": {
            "fileName": "src/components/Checkbox/Checkbox.tsx",
            "name": "CheckboxProps"
          },
          "required": false,
          "type": {
            "name": "\"black\" | \"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
          }
        },
        "name": {
          "defaultValue": null,
          "description": "The name of the checkbox. Mirrors the name HTML attribute.",
          "name": "name",
          "parent": {
            "fileName": "src/components/Checkbox/Checkbox.tsx",
            "name": "CheckboxProps"
          },
          "required": false,
          "type": { "name": "string | undefined" }
        },
        "disabled": {
          "defaultValue": null,
          "description": "",
          "name": "disabled",
          "parent": {
            "fileName": "src/components/Checkbox/Checkbox.tsx",
            "name": "CheckboxProps"
          },
          "required": false,
          "type": { "name": "boolean | undefined" }
        }
      },
      "ext_com_draculaui_docgen": {
        "description": "Checkbox is a styled HTML Input of type checkbox.\n\nThere are no behavior changes applied to the native HTML tag other\nthan light styling done via CSS in order to keep check boxes accessible.",
        "displayName": "Checkbox",
        "methods": [],
        "props": {
          "color": {
            "defaultValue": null,
            "description": "The Dracula UI theme color for the checkbox.",
            "name": "color",
            "parent": {
              "fileName": "src/components/Checkbox/Checkbox.tsx",
              "name": "CheckboxProps"
            },
            "required": false,
            "type": {
              "name": "\"black\" | \"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
            }
          },
          "name": {
            "defaultValue": null,
            "description": "The name of the checkbox. Mirrors the name HTML attribute.",
            "name": "name",
            "parent": {
              "fileName": "src/components/Checkbox/Checkbox.tsx",
              "name": "CheckboxProps"
            },
            "required": false,
            "type": { "name": "string | undefined" }
          },
          "disabled": {
            "defaultValue": null,
            "description": "",
            "name": "disabled",
            "parent": {
              "fileName": "src/components/Checkbox/Checkbox.tsx",
              "name": "CheckboxProps"
            },
            "required": false,
            "type": { "name": "boolean | undefined" }
          }
        }
      }
    }
  ]
}
