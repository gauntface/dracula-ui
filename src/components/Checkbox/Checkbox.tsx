import React, { HTMLAttributes } from 'react'

export const checkboxColors = {
  black: 'drac-checkbox-black',
  white: 'drac-checkbox-white',
  cyan: 'drac-checkbox-cyan',
  green: 'drac-checkbox-green',
  orange: 'drac-checkbox-orange',
  pink: 'drac-checkbox-pink',
  purple: 'drac-checkbox-purple',
  red: 'drac-checkbox-red',
  yellow: 'drac-checkbox-yellow'
}

export interface CheckboxProps extends HTMLAttributes<HTMLInputElement> {
  color: keyof typeof checkboxColors
  name?: string
}

export const Checkbox: React.FC<CheckboxProps> = props => {
  const finalProps = {
    ...props,
    className: `drac-checkbox ${checkboxColors[props.color]}`
  }

  return <input type="checkbox" {...finalProps} />
}
