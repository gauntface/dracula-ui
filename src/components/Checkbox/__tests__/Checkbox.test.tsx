import React from 'react'
import * as ReactDOM from 'react-dom'
import { Checkbox } from '../Checkbox'

describe('Checkbox', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Checkbox color="white" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})
