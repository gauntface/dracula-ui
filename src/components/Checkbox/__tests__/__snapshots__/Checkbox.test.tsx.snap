// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Checkbox Basic Usage 1`] = `
Object {
  "docs": "",
  "html": "<div>
  <input
    type=\\"checkbox\\"
    id=\\"one\\"
    color=\\"purple\\"
    name=\\"demo\\"
    class=\\"drac-checkbox drac-checkbox-purple\\"
    checked=\\"\\"
  /><label for=\\"one\\" class=\\"drac-text drac-text-white\\">Checked</label>
</div>
",
  "react": "<div>
  <Checkbox id=\\"one\\" defaultChecked={true} color=\\"purple\\" name=\\"demo\\" />
  <label htmlFor=\\"one\\" className=\\"drac-text drac-text-white\\">
    Checked
  </label>
</div>;
",
  "title": "Usage",
}
`;

exports[`Site: Checkbox Colors 1`] = `
Object {
  "docs": "",
  "html": "<div style=\\"display: flex; flex-direction: column\\">
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"white\\"
      color=\\"white\\"
      name=\\"demo\\"
      class=\\"drac-checkbox drac-checkbox-white\\"
      checked=\\"\\"
    /><label for=\\"white\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"cyan\\"
      color=\\"cyan\\"
      name=\\"demo\\"
      class=\\"drac-checkbox drac-checkbox-cyan\\"
      checked=\\"\\"
    /><label for=\\"cyan\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"green\\"
      color=\\"green\\"
      name=\\"demo\\"
      class=\\"drac-checkbox drac-checkbox-green\\"
      checked=\\"\\"
    /><label for=\\"green\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"orange\\"
      color=\\"orange\\"
      name=\\"demo\\"
      class=\\"drac-checkbox drac-checkbox-orange\\"
      checked=\\"\\"
    /><label for=\\"orange\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"pink\\"
      color=\\"pink\\"
      name=\\"demo\\"
      class=\\"drac-checkbox drac-checkbox-pink\\"
      checked=\\"\\"
    /><label for=\\"pink\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"purple\\"
      color=\\"purple\\"
      name=\\"demo\\"
      class=\\"drac-checkbox drac-checkbox-purple\\"
      checked=\\"\\"
    /><label for=\\"purple\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"red\\"
      color=\\"red\\"
      name=\\"demo\\"
      class=\\"drac-checkbox drac-checkbox-red\\"
      checked=\\"\\"
    /><label for=\\"red\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"yellow\\"
      color=\\"yellow\\"
      name=\\"demo\\"
      class=\\"drac-checkbox drac-checkbox-yellow\\"
      checked=\\"\\"
    /><label for=\\"yellow\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
</div>
",
  "react": "<div style={{ display: \\"flex\\", flexDirection: \\"column\\" }}>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"white\\" defaultChecked={true} color=\\"white\\" name=\\"demo\\" />
    <label htmlFor=\\"white\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"cyan\\" defaultChecked={true} color=\\"cyan\\" name=\\"demo\\" />
    <label htmlFor=\\"cyan\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"green\\" defaultChecked={true} color=\\"green\\" name=\\"demo\\" />
    <label htmlFor=\\"green\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"orange\\" defaultChecked={true} color=\\"orange\\" name=\\"demo\\" />
    <label htmlFor=\\"orange\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"pink\\" defaultChecked={true} color=\\"pink\\" name=\\"demo\\" />
    <label htmlFor=\\"pink\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"purple\\" defaultChecked={true} color=\\"purple\\" name=\\"demo\\" />
    <label htmlFor=\\"purple\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"red\\" defaultChecked={true} color=\\"red\\" name=\\"demo\\" />
    <label htmlFor=\\"red\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"yellow\\" defaultChecked={true} color=\\"yellow\\" name=\\"demo\\" />
    <label htmlFor=\\"yellow\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
</div>;
",
  "title": "Colors",
}
`;

exports[`Site: Checkbox Disabled 1`] = `
Object {
  "docs": "",
  "html": "<div style=\\"display: flex; flex-direction: column\\">
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"one\\"
      disabled=\\"\\"
      color=\\"white\\"
      name=\\"demo\\"
      class=\\"drac-checkbox drac-checkbox-white\\"
      checked=\\"\\"
    /><label for=\\"one\\" class=\\"drac-text drac-text-white\\">Checked</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"two\\"
      disabled=\\"\\"
      color=\\"white\\"
      name=\\"demo\\"
      class=\\"drac-checkbox drac-checkbox-white\\"
    /><label for=\\"two\\" class=\\"drac-text drac-text-white\\">Unchecked</label>
  </div>
</div>
",
  "react": "<div style={{ display: \\"flex\\", flexDirection: \\"column\\" }}>
  <div style={{ marginBottom: 10 }}>
    <Checkbox
      id=\\"one\\"
      disabled={true}
      defaultChecked={true}
      color=\\"white\\"
      name=\\"demo\\"
    />
    <label htmlFor=\\"one\\" className=\\"drac-text drac-text-white\\">
      Checked
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"two\\" disabled={true} color=\\"white\\" name=\\"demo\\" />
    <label htmlFor=\\"two\\" className=\\"drac-text drac-text-white\\">
      Unchecked
    </label>
  </div>
</div>;
",
  "title": "Disabled",
}
`;
